modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -922,6 +922,146 @@ GAPI_FLUID_KERNEL(GFluidNot, cv::gapi::core::GNot, false)
     }
 };
 
+//--------------------------------------
+//
+// Fluid math kernels: bitwise with Scalar
+//
+//--------------------------------------
+
+static void checkScalarForBitwise(const cv::Scalar &_scalar, const int scalarI[4])
+{
+    if (!((_scalar[0] == scalarI[0]) && (_scalar[1] == scalarI[1]) &&
+          (_scalar[2] == scalarI[2]) && (_scalar[3] == scalarI[3])))
+    {
+        CV_Error(cv::Error::StsBadArg, "Bitwise operations make sense with integral types only");
+    }
+}
+
+template<typename DST>
+static inline DST bw_andS(DST x, int y)
+{
+    return x & saturate<DST>(y);
+}
+
+template<typename DST>
+static inline DST bw_orS(DST x, int y)
+{
+    return x | saturate<DST>(y);
+}
+
+template<typename DST>
+static inline DST bw_xorS(DST x, int y)
+{
+    return x ^ saturate<DST>(y);
+}
+
+// manually unroll the inner cycle by channels
+// (reuse arithmetics function above of the same purpose)
+template<typename DST, typename FUNC>
+static inline void run_bitwise_s(DST out[], const DST in[], int width, int chan,
+                                 const int scalar[4], FUNC func)
+{
+    run_arithm_s(out, in, width, chan, scalar, func);
+}
+
+template<typename DST, typename SRC>
+static void run_bitwise_s(Buffer &dst, const View &src, const int scalar[4], Bitwise bitwise)
+{
+    static_assert(std::is_same<DST, SRC>::value, "wrong types");
+
+    const auto *in  = src.InLine<SRC>(0);
+          auto *out = dst.OutLine<DST>();
+
+    int width  = dst.length();
+    int chan   = dst.meta().chan;
+
+    switch (bitwise)
+    {
+    case BW_AND:
+        run_bitwise_s(out, in, width, chan, scalar, bw_andS<DST>);
+        break;
+    case BW_OR:
+        run_bitwise_s(out, in, width, chan, scalar, bw_orS<DST>);
+        break;
+    case BW_XOR:
+        run_bitwise_s(out, in, width, chan, scalar, bw_xorS<DST>);
+        break;
+    default: CV_Error(cv::Error::StsBadArg, "unsupported bitwise operation");
+    }
+}
+
+GAPI_FLUID_KERNEL(GFluidAndS, cv::gapi::core::GAndS, false)
+{
+    static const int Window = 1;
+
+    static void run(const View &src, const cv::Scalar &_scalar, Buffer &dst)
+    {
+
+        const int scalar[4] = {
```

Comment:
Do you really need to do this? If so, please, make it a function since you're using it below several times

Reply:
Yeah, I think this should be done.

I changed the function `checkScalar` -> `convertScalar`; but I'm afraid the code I wrote is not secure. Therefore, I didn't find anything better yet. Can you help me with suggestions, please?

Reply:
Can you refer the line here?

Reply:
OK I started a new conversation about this function